juju/juju
# fix: remove instance data checks on caas provisioning

A wrong set of checks was incorrectly added on
https://github.com/juju/juju/pull/18165, which added checks on instance data and therefore machine retrieval for caas apps provisioning. This returns an error when trying to deploy.






## QA steps

Deploying on microk8s should test this codepath:

```sh
$ make microk8s-operator-update
$ juju bootstrap microk8s c
$ juju add-model m
```
Workaround the model config bug:
```sh
$ juju model-config workload-storage="microk8s-hostpath"
```
Then deploy:
```
$ juju deploy postgresql-k8s --trust
```
Application should reach a idle state and no errors observed on debug-log.

## Links


**Jira card:** JUJU-
